Open Judge System/OJS.Services/Business/Problems/ProblemsBusinessService.cs

```diff
@@ -38,5 +68,57 @@ public void RetestById(int id)
                 scope.Complete();
             }
         }
+
+        public void DeleteById(int id)
+        {
+            this.problemResourcesData.DeleteByProblem(id);
+
+            this.testRunsData.DeleteByProblem(id);
+
+            this.testsData.DeleteByProblem(id);
+
+            this.submissionsData.DeleteByProblem(id);
+
+            this.problemsData.DeleteById(id);
+
+            if (!this.problemsData.IsFromOnlineContestById(id))
+            {
+                this.problemGroupsBusiness.DeleteByProblem(id);
+            }
+        }
+
+        public void DeleteByContest(int contestId)
+        {
+            this.problemResourcesData.DeleteByContest(contestId);
+
+            this.testRunsData.DeleteByContest(contestId);
+
+            this.testsData.DeleteByContest(contestId);
```

Comment:
Why should all these entities be deleted as well? That would make restoring a deleted problem pretty useless, what if someone deletes a problem by mistake. Problems are deletable entities, there is no need to delete all entities that rely on them.